tests/models/test_amp.py

```diff
@@ -23,19 +23,34 @@
 from pytorch_lightning.plugins.environments import SLURMEnvironment
 from pytorch_lightning.trainer.states import TrainerState
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
-from tests.helpers import BoringModel
+from tests.helpers import BoringModel, RandomDataset
 from tests.helpers.runif import RunIf
 
 
 class AMPTestModel(BoringModel):
 
-    def training_step(self, batch, batch_idx):
+    def _step(self, batch, batch_idx):
         assert torch.is_autocast_enabled()
         output = self(batch)
         assert output.dtype == torch.float16
         loss = self.loss(batch, output)
         return {"loss": loss}
 
+    def training_step(self, batch, batch_idx):
+        return self._step(batch, batch_idx)
+
+    def validation_step(self, batch, batch_idx):
+        self._step(batch, batch_idx)
+
+    def test_step(self, batch, batch_idx):
+        self._step(batch, batch_idx)
```

Comment:
You are missing the return statements
Boring model has epoch_end methods implemented that want the outputs

Reply:
this will probably fix the tests

Reply:
```suggestion
    def validation_step(self, batch, batch_idx):
        return self._step(batch, batch_idx)

    def test_step(self, batch, batch_idx):
        return self._step(batch, batch_idx)
```

Reply:
The epoch_end methods require it.
You can also do `model.validation_epoch_end = None` I suppose
In any case, in order to make the tests pass something has to change

Reply:
my intuition was wrong, adding the return is not enough. The training step returns a key "loss" but the validation epoch end hooks try to access a key "x" returned from validation_step.